src/java.base/share/classes/sun/launcher/SecuritySettings.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package sun.launcher;
+
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSocket;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.security.NoSuchAlgorithmException;
+import java.security.Provider;
+import java.security.Security;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Properties;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import jdk.internal.access.SharedSecrets;
+import static sun.launcher.LauncherHelper.INDENT;
+import static sun.launcher.LauncherHelper.TWOINDENT;
+
+/**
+ * A utility class for security libs functionality
+ * in the -XshowSettings:security output
+ */
+public final class SecuritySettings {
+
+    private static final String THREEINDENT = TWOINDENT + INDENT;
+    private static final String PROV_INFO_STRING = "Provider information: ";
+    private static PrintStream ostream = null;
+
+    static void printSecuritySettings(String arg, PrintStream stream) {
+        ostream = stream;
+        switch (arg) {
+            case "properties" -> printSecurityProperties();
+            case "providers"  -> printSecurityProviderConfig(true);
+            case "tls"        -> printSecurityTLSConfig(true);
+            case "all"        -> printAllSecurityConfig();
+            default           -> {
+                ostream.println("Unrecognized security subcommand. See \"java -X\" for help");
+                ostream.println("Printing all security settings");
+                printAllSecurityConfig();
```

Comment:
The error message is going to get lost in the volume of settings.
Allowing bad command input reinforces learning the wrong suboption; though it may duplicate the help, I'd print the allowed options.

Reply:
@RogerRiggs  - do you mean to print nothing in the "bad command input" scenario ? The current -XshowSettings launch behaviour prints all data if a bad value is passed to it. I was mimicking this for security subcommands. 

Are you suggesting to print the relevant part of the "java -X" help menu here or is restoring the message with value subcommands sufficient ?

Reply:
Restoring the message with value subcommands is what I would prefer.
I do disagree with the original choice of printing everything if the command is wrong and would not propagate that behavior to the new command.  
Separately, I would consider changing the existing behavior but I suspect that would fail to achieve consensus based on compatibility concerns.

Reply:
Fair point. Modified the code so that we don't print any security settings if a bad security subcommand value is specified. Info message instead. I'll update the CSR on that point also.

Agreed on the compatibility concerns of changing the existing behaviour already present in this area.